src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -248,6 +291,74 @@ public string ToString(string directory)
             return string.Join(System.Environment.NewLine, lines.ToArray());
         }
 
+        /// <summary>
+        /// Returns the string representation of the match object same format as ToString()
+        /// and adds color to the matched text if virtual terminal is supported.
+        /// </summary>
+        /// <param name="directory">Directory to use as the root when calculating the relative path.</param>
+        /// <returns>The colored string representation of the match object.</returns>
+        public string ToEmphasizedString(string directory)
+        {
+            if (!Emphasize || ! _supportsVirtualTerminal) {
+                return ToString(directory);
+            }
+
+            string originalLine = Line;
+            string open = "\u001b[7m";
+            string close = "\u001b[0m";
+
+            Line = string.Create((_matchIndexes.Count * (open.Length + close.Length)) + Line.Length, Line, (chars, buf) =>
+            {
+                int lineIndex = 0;
+                int charsIndex = 0;
+                for (int i = 0; i < _matchIndexes.Count; i++) 
+                {
+                    // Adds characters before match
+                    while (lineIndex < _matchIndexes[i])
+                    {
+                        chars[charsIndex] = Line[lineIndex];
+                        lineIndex++;
+                        charsIndex++;
+                    }
+                    
+                    // Adds opening vt sequence
+                    for (int j = 0; j < open.Length; j++)
+                    {
+                        chars[charsIndex] = open[j];
+                        charsIndex++;
+                    }
+                    
+                    // Adds characters being emphasized
+                    for (int j = 0; j < _matchLengths[i]; j++)
+                    {
+                        chars[charsIndex] = Line[lineIndex];
+                        lineIndex++;
+                        charsIndex++; 
+                    }
+
+                    // Adds closing vt sequence
+                    for (int j = 0; j < close.Length; j++)
+                    {
+                        chars[charsIndex] = close[j];
+                        charsIndex++;
+                    }
+                }
+
+                // Adds remaining characters in line
+                while (lineIndex < Line.Length)
+                {
+                    chars[charsIndex] = Line[lineIndex];
+                    lineIndex++;
+                    charsIndex++;
+                }
+            });
+
+            string modifiedLine = ToString(directory);
```

Comment:
The way `Line` is used here is a bit opaque I think. It might be that a better approach is to create a private overload of `ToString()` and put all the current `ToString(directory)` logic into it like:

```csharp
public string ToString(string directory)
{
    return ToString(Line, directory);
}

private string ToString(string line, string directory)
{
    ... // actual content with 'Line' changed to 'line'
}
```

Then for the method you'd want:
```csharp
public string ToEmphasizedString(string directory)
{
    if (!_emphasize)
    {
        return ToString(directory);
    }

    string emphasizedLine = EmphasizeLine();
    return ToString(emphasizedLine, directory);
}

private string EmphasizeLine()
{
    const string invertColorsVT100 = "\u001b[7m";
    const string resetVT100 = "\u001b[0m";
    return string.Create((_matchIndexes.Count * (open.Length + close.Length)) + Line.Length, Line, (chars, buf) => {
        // Complicated logic
    });
}
```